addons/user-id/userscript.js

```diff
@@ -0,0 +1,18 @@
+export default async function ({ addon, global, console }) {
+// define the user's username header
+let usernameheader = document.getElementsByClassName("header-text")[0].firstElementChild
+
+let userid = document.createElement("font");
+userid.style.color = "lightgrey"
+userid.style.size = 2;
+
+// fetch the user's id by making a get request to api.scratch.mit.edu
+// location.pathname returns /users/USERNAME/
+fetch('https://api.scratch.mit.edu' + location.pathname)
+  .then(response => response.json())
+  // Add the ID of the response to the username header
+  .then(data => userid.textContent = "#" + data.id);
+
+// add the id next to the header
+usernameheader.appendChild(userid);
```

Comment:
```suggestion
export default async function ({ addon }) {
	let usernameHeader = document.querySelector("header-text").firstElementChild;
	const userId = Object.assign(document.createElement("font"), {
		innerText: `#${Scratch.INIT_DATA.PROFILE.model.userId}`,
		className: "sa-user-id"
	});
	userId.style.color = "lightgrey";
	userId.style.size = 2;
	
	usernameHeader.appendChild(userId)
```

Reply:
why

Reply:
it doesnt include requests, it uses data built in to scratch's profile page

Reply:
Uncaught (in promise) TypeError: Cannot read property 'firstElementChild' of null

Reply:
oops its `document.querySelector(".header-text").firstElementChild`

Reply:
how is that different